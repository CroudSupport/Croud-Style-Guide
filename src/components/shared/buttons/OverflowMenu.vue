<template>
  <button ref="options" class="ui dropdown icon pointing top left circular basic button">
    <i class="ellipsis vertical icon"></i>
    <div class="ui menu">
        <slot name="options">
            <div
                v-for="option in options"
                class="item"
                :class="{disabled: option.disabled}"
                :key="option.name"
                @click="option.method"
            >
                <i v-if="option.icon" class="icon" :class="option.icon"></i>
                {{ option.name }}
            </div>
        </slot>
    </div>
  </button>
</template>

<script>
export default {
    props: {
        options: {
            type: Array,
            default: [],
        },
    },

    mounted() {
        $(this.$refs.options).dropdown()
    },
}
</script>
