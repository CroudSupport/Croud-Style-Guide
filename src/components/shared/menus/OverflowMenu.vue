<template>
    <button ref="options" class="ui dropdown icon pointing top circular button"
    :class="{left, right: !left, 'no-pointer': noPointer}">
        <i class="ellipsis vertical icon"></i>
        <div class="ui menu">
            <slot name="options">
                <template v-for="(option, index) in options" :key="index">
                    <template v-if="!option.method">
                        <div v-if="index > 0" class="divider"></div>
                        <div v-if="option.name" class="header">
                            <i v-if="option.icon" class="icon" :class="option.icon"></i>
                            {{ option.name }}
                        </div>
                    </template>
                    <div v-else
                        class="item"
                        :class="{ disabled: option.disabled}"
                        :key="option.name"
                        @click="option.method"
                    >
                        <i v-if="option.icon" class="icon" :class="option.icon"></i>
                        {{ option.name }}
                    </div>
                </template>
            </slot>
        </div>
    </button>
</template>

<script>
/**
* Overflow Menu component
* @example ./croud-overflow-menu.md
*/

export default {
    name: 'croud-overflow-menu',
    props: {
        /**
        * Menu item information
        */
        options: {
            type: Array,
            default: [],
        },
        /**
        * Align menu to left
        */
        left: {
            type: Boolean,
            default: false,
        },
        /**
        * Hide pointer in Dropdown
        */
        noPointer: {
            type: Boolean,
            default: false,
        },
    },
    mounted() {
        $(this.$refs.options).dropdown({
            transition: 'slide down',
            action: 'hide',
        })
    },
}
</script>

<style lang="scss" scoped>
    @import '../../../resources/sass/variables/_all.scss';

    .ui.dropdown{
        &.button{
            background: none;
            &.active.visible {
                &:hover{
                    background: none;
                }
            }

            &:focus{
                background: none!important;
            }

            .icon{
                font-size: $croud-font-size-large;
            }

            &.no-pointer{
                .menu{
                   &::after{
                        display: none;
                   }
                }
            }
            .menu{
                position: absolute;
                right:0;
                margin-top: -.3em !important;
                border-radius: 2px;
                box-shadow: 0 1px 2px .5px rgba($croud-colour-black, .4);
                border: none;
                min-width: 112px;
                padding: 8px 0;

               .divider {
                   margin: 0;
               }

                &::after{
                    box-shadow: none;
                    border-left: 1px solid rgba($croud-colour-black, .15);
                    border-top: 1px solid rgba($croud-colour-black,.15)
                }


                @media screen and (min-width: 450px){
                    min-width: 128px;
                    max-width: 156px;
                }
                .item{
                    display: flex;
                    align-items: center;
                    font-size: $croud-font-size-small;
                    border-radius: 0;
                    color: $croud-colour-grey-dark;
                    padding-left: 16px!important;
                    height: 48px;

                    .icon{
                        margin-left: -1em;
                        font-size: $croud-font-size-small;
                        opacity: .8;
                    }

                    @media screen and (min-width: 450px){
                        height: 40px;
                        padding-left: 24px!important;
                    }
                }
            }
        }
    }

</style>
