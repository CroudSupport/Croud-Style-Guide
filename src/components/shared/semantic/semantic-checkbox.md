### semantic-checkbox
For our checkboxes, we use the semantic-checkbox component from the [croud-vue-semantic](https://github.com/CroudSupport/vue-semantic) library. Below are a few examples of semantic-checkbox in action. Click [here](http://croudsupport.github.io/vue-semantic/#semantic-checkbox) for full component docs.

### Basic usage

    <semantic-checkbox v-model="checkbox" label="🍕 Pizza" />

### Styling
Pass through the **type** prop to change the style of checkbo

    <div class="ui form">
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Checkbox" />
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Slider" type="slider"/>
        </div>
        <div class="ui field">
            <semantic-checkbox v-model="checkbox" label="Toggle" type="toggle" />
        </div>
    </div>

### Disabled

    <semantic-checkbox label="🍕 Pizza" type="toggle" :disabled="true"/>
