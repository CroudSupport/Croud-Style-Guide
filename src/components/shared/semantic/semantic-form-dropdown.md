### semantic-form-dropdown
For our form dropdowns, we use the semantic-form-dropdown component from the [croud-vue-semantic](https://github.com/croudtech/vue-semantic) library. Below are a few examples of semantic-form-dropdown in action. Click [here](http://croudtech.github.io/vue-semantic/#semantic-form-dropdown) for full component docs.

### Basic usage

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]"></semantic-form-dropdown>

### Styling
You can use the props to change the styling of the dropdown

    <div class="ui form">
        <div class="field">
            <label>Fluid</label>
            <semantic-form-dropdown fluid v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>

        <div class="three fields">
            <div class="field">
                <label>Search</label>
                <semantic-form-dropdown search v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Multiple</label>
                <semantic-form-dropdown multiple v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
            <div class="field">
                <label>Disabled</label>
                <semantic-form-dropdown :disabled="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
            </div>
        </div>

        <div class="inline field">
            <label>Inline</label>
            <semantic-form-dropdown :inline="true" v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" />
        </div>
    </div>

### Using slots
The **default** slot allows you to add custom options to the top of the dropdown

    <semantic-form-dropdown v-model="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]">
        <div class="item" v-data="3">
            <b>Baz</b>
        </div>
    </semantic-form-dropdown>

### Handle Events
This component emits the **dropdown-selected** event when an item is selected.

    <semantic-form-dropdown :value="dropdownVal" :options="[{id: 1, name: 'Foo'},{id: 2, name: 'Bar'}]" @dropdown-selected="$toasted.success('TaDa!!!')"></semantic-form-dropdown>
