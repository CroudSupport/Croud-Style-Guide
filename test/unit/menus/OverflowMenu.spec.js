import Vue from 'vue'
import OverflowMenu from '../../../src/components/shared/menus/OverflowMenu'
import '../../../semantic/dist/semantic'

const Constructor = Vue.extend(OverflowMenu)

const vm = new Constructor({
    propsData: {
        options: [
            {
                name: 'Delete',
                method: () => {},
            },
            {
                name: 'Postpone',
                method: () => {},
            },
            {
                name: 'Copy',
                method: () => {},
                disabled: true,
            },
        ],
    },
}).$mount()

describe('OverflowMenu', () => {
    it('should match the snapshot', () => {
        expect(vm.$el).toMatchSnapshot()
    })
})
