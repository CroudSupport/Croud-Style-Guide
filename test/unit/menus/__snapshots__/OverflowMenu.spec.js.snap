// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overflow Menu should match the snapshot 1`] = `
<button
  class="ui dropdown icon pointing top circular button right"
  tabindex="0"
>
  <i
    class="ellipsis vertical icon"
  />
  <div
    class="ui menu"
    tabindex="-1"
  >
    <div
      class="item"
    >
      Delete
    </div>
    <div
      class="item"
    >
      Postpone
    </div>
    <div
      class="item disabled"
    >
      Copy
    </div>
  </div>
</button>
`;
